homeassistant/components/alarm_control_panel/egardia.py

```diff
@@ -71,41 +75,63 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     rs_enabled = config.get(CONF_REPORT_SERVER_ENABLED)
     rs_port = config.get(CONF_REPORT_SERVER_PORT)
     rs_codes = config.get(CONF_REPORT_SERVER_CODES)
+    version = config.get(CONF_VERSION)
 
     try:
         egardiasystem = egardiadevice.EgardiaDevice(
-            host, port, username, password, '')
+            host, port, username, password, '', version)
     except requests.exceptions.RequestException:
         raise exc.PlatformNotReady()
     except egardiadevice.UnauthorizedError:
         _LOGGER.error("Unable to authorize. Wrong password or username")
         return False
 
-    add_devices([EgardiaAlarm(
-        name, egardiasystem, hass, rs_enabled, rs_port, rs_codes)], True)
+    eg_dev = EgardiaAlarm(
+        name, egardiasystem, rs_enabled, rs_codes)
+
+    if rs_enabled:
+        # Set up the egardia server
+        _LOGGER.info("Setting up EgardiaServer")
+        try:
+            if D_EGARDIASRV not in hass.data:
+                hass.data[D_EGARDIASRV] = egardiaserver.EgardiaServer('',
```

Comment:
Cache the server instance in a local variable so you don't need to look that up in `hass.data` every time.
```py
hass.data[D_EGARDIASRV] = server = egardiaserver.EgardiaServer(...)
```